### PR TITLE
AB#64775_support_for_reference_data 

### PR DESCRIPTION
**Pull request by MwanPygmay**
# Description

* Ability to setup charts using reference data as datasource
* Representation of objects and arrays on summary cards 

## Ticket

https://dev.azure.com/WHOHQ/EMSSAFE/_workitems/edit/64775/

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (refactor or addition to existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

* Created a form with checkbox and dropdown question with choices coming from ref data and added those to summary cards
image
* Setup a graph with datasource being refData (see bellow)

## Screenshots
Peek 2023-08-30 03-01

# Checklist:

( * == Mandatory ) 

- [x] * My code follows the style guidelines of this project
- [x] * Linting does not generate new warnings
- [x] * I have performed a self-review of my own code
- [x] * I have commented my code, particularly in hard-to-understand areas
- [x] * I have put JSDoc comment in all required places
- [x] * My changes generate no new warnings
- [x] * I have included screenshots describing my changes if relevant
- [x] * I have selected labels in the Pull Request, according to the changes with code brings
- [ ] I have made corresponding changes to the documentation ( if required )
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: libs/safe/src/lib/components/widgets/summary-card-settings/summary-card-general/summary-card-general.component.ts [hunk 1]
```diff
@@ -36,7 +36,7 @@ import {
 import { Dialog } from '@angular/cdk/dialog';
 
 /** Default number of resources to be fetched per page */
-const ITEMS_PER_PAGE = 10;
+const ITEMS_PER_PAGE = 30;
```
Comment (RenzoPrats) [inline libs/safe/src/lib/components/widgets/summary-card-settings/summary-card-general/summary-card-general.component.ts]: This should be 10, no?
```suggestion
const ITEMS_PER_PAGE = 10;
```

**Review by RenzoPrats — APPROVED** [review]
Working fine, looks good to me!

**Comment by AntoineRelief** [comment]
adding on top of that:
- perhaps list the available fields in the tinymce editor
- check that the code is actually working fine
- check if working for when we use ref data as tagbox & dropdown

**Comment by AntoineRelief** [comment]
@matheus-relief 
I think you over complexified the pr 😅

it's not what I asked
I was just expecting the resources that use ref data to be usable, not the ref data to be usable in charts..

**Comment by AntoineRelief** [comment]
and what I meant for the reference data is text widget is that:
Capture d’écran 2023-12-06 à 10 44 36

and not that:

Capture d’écran 2023-12-06 à 10 44 15

Sorry if that was not clear

I think you should check from your past commits @matheus-relief what is fine or not, remove them, and do a force push to erase previous commits

**Comment by AntoineRelief** [comment]
putting back as draft based on previous comments

**Pull request merged**